src/main/java/org/elasticsearch/percolator/PercolatorService.java

```diff
@@ -789,7 +796,15 @@ public PercolateShardResponse doPercolate(PercolateShardRequest request, Percola
     private void queryBasedPercolating(Engine.Searcher percolatorSearcher, PercolateContext context, QueryCollector percolateCollector) throws IOException {
         Filter percolatorTypeFilter = context.indexService().mapperService().documentMapper(TYPE_NAME).typeFilter();
         percolatorTypeFilter = context.indexService().cache().filter().cache(percolatorTypeFilter);
-        XFilteredQuery query = new XFilteredQuery(context.percolateQuery(), percolatorTypeFilter);
+
+        final Filter filter;
+        if (context.aliasFilter() != null) {
+            filter = new AndFilter(Arrays.asList(context.aliasFilter(), percolatorTypeFilter));
```

Comment:
order matters here right? wondering if type shouldn't be first, given that we don't know how expensive the alias filter will be since it's user provided. Thoughts?